### ISSUE TITLE
False positive SING_SINGLETON_GETTER_NOT_SYNCHRONIZED with eager instances

### ISSUE THREAD

**Issue by rovarga** [spotbugs/spotbugs#2932]
The following code:
```
  public static final Foo {
    private static final Foo INSTANCE = new Foo();

    private Foo() {
        // hidden on purpose
    }

    public static Foo instance() {
        return INSTANCE
    }
 }
```

triggers a SING_SINGLETON_GETTER_NOT_SYNCHRONIZED (at https://github.com/opendaylight/yangtools/blob/master/common/yang-common/src/main/java/org/opendaylight/yangtools/yang/common/Empty.java#L40 in my case).

 I believe this is a false positive: INSTANCE is guaranteed to be initialized before a call to Foo.instance() can be made. It is therefore superfluous to require synchronized() on instance() -- and in fact doing so creates a synchronization hotspot, leading to abysmal scalability.

**Comment by JuditKnoll** [comment]
Thank you for reporting this issue. I'll will open a PR fixing this soon. I just want to include the fix to https://github.com/spotbugs/spotbugs/issues/2934 in the PR, which is connected to the same detector.

**Comment by vy** [comment]
Unaware of this ticket, [posted the very same question to SO](https://stackoverflow.com/q/78326620/1278899) and [contacted to CMU SEI](https://www.sei.cmu.edu/contact-us/) asking for explanation on the related part of [the SEI CERT MSC07-J rule](https://wiki.sei.cmu.edu/confluence/display/java/MSC07-J.+Prevent+multiple+instantiations+of+singleton+objects).
